### PR TITLE
Add gen_platform config param

### PR DESCRIPTION
**Pull request by karlkfi**
`platform` indicates WHAT platform DC/OS is deployed on, rather than HOW the install artifacts were generated (`provider`)

### CODE REVIEW

**Comment by malnick** [comment]
It looks like Velocity is stalled...

**Comment by karlkfi** [comment]
`velocity` was renamed `jenkins` with the switch from "product" to "solution". I updated the webhooks and resent the lost messages from the last few PRs.

**Comment by karlkfi** [comment]
FWIW, we will likely want to expose both platform and provider via some API endpoint in the future, which likely doesn't make sense to be in dcos-signal. It would seem to make sense to have it be some component that is the source of truth for the dcos configuration. That way people could retrieve the config they put in. This would be useful for tests, debugging, reproducing environments, recovering from lost install artifacts, etc.

File: config/config_test.go [hunk 1]
```diff
@@ -93,7 +93,8 @@ func TestGetExternalConfig(t *testing.T) {
 	noEntConfig := []byte(`
 {
 "customer_key": "",
-"gen_provider": "onprem"	
+"gen_platform": "vagrant-virtualbox",
```
Comment (malnick) [inline config/config_test.go]: What if the user is using Fusion or another vbox provider? Do we really want to hard code this?

**Pull request merged**